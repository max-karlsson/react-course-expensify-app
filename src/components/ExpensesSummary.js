import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
  const formattedExpensestotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        {expenseCount > 0 && `Viewing ${expenseCount} ${expenseWord} totalling ${formattedExpensestotal}`}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);