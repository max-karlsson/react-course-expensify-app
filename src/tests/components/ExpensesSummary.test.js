import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={333}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={33334244424}/>);
  expect(wrapper).toMatchSnapshot();
});