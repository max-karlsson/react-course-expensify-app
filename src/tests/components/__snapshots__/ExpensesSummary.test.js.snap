// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $3.33
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 23 expenses totalling $333,342,444.24
  </h1>
</div>
`;
